cli/cmd/test_connection_test.go

```diff
@@ -52,3 +53,25 @@ func TestTestConnection(t *testing.T) {
 		})
 	}
 }
+
+// errorsContainAny checks if any error in a slice contains at least one of the substrings.
+func errorsContainAny(errs []error, substrings []string) bool {
```

Comment:
What's the benefit of these helpers instead of doing `require.ErrorContains` in a loop like before? Don't we want to test that the first substring appeared in the first error, and the second substring in the second error?

Reply:
We have multiple errors from `Unwrap() []error` so the previous code won't work. We can require the same count between `len(resultOfUnWrap)` and `len(want)` and try matching them in order, but that means the test expects a certain order in the results, which it currently doesn't.

Reply:
Doesn't that mean that if cloudflare errors out but postgres not the test will still pass? Also should we check a better substring than the plugin name? Something like "test failed"?

Reply:
If the plugin name is not in the list of _errors_ then it's safe to assume the test succeeded. We could also embed the whole result in the failure error, let me do that so it's simpler.

Reply:
Refactored in https://github.com/cloudquery/cloudquery/pull/17993/commits/7991515a124020497ec7273ce834bf1a64e62aa4 , CLI output doesn't change.